crates/bitwarden-auth/src/registration.rs

```diff
@@ -33,4 +45,129 @@ impl RegistrationClient {
         let api_client = &client.get_api_configurations().await.api_client;
         // Do API request here. It will be authenticated using the client's tokens.
     }
+
+    /// Initializes a new cryptographic state for a user and posts it to the server; enrolls in
+    /// admin password reset and finally enrolls the user to TDE unlock.
+    pub async fn post_keys_for_tde_registration(
+        &self,
+        org_id: String,
+        org_public_key: B64,
+        // Note: Ideally these would be set for the register client, however no such functionality
+        // exists at the moment
+        user_id: String,
+        device_id: String,
+        trust_device: bool,
+    ) -> Result<TdeRegistrationResult, UserRegistrationError> {
+        let client = &self.client.internal;
+        #[allow(unused_variables)]
+        let api_client = &client.get_api_configurations().await.api_client;
+        let user_id =
+            UserId::from_str(user_id.as_str()).map_err(|_| UserRegistrationError::Serialization)?;
+
+        // First call crypto API to get all keys
+        info!("Initializing account cryptography");
+        let (
+            cryptography_state,
+            user_key,
+            account_cryptographic_state_request,
+            device_key_set,
+            reset_password_key,
+        ) = self
+            .client
+            .crypto()
+            .make_user_tde_registration(user_id, org_public_key)
+            .map_err(|_| UserRegistrationError::Crypto)?;
```

Comment:
❓ :  This swallows the actual error - is there value in exposing it in a log or is that dangerous?